common/src/main/java/ch/admin/bag/covidcertificate/common/qr/QRCodeReaderHelper.kt

```diff
@@ -69,8 +63,13 @@ object QRCodeReaderHelper {
 						page.renderToBitmap(page.width, page.height).use {
 							yield(it)
 						}
-						page.renderToBitmap(pixelWidth, pixelHeight).use {
-							yield(it)
+
+						if (!memoryInfo.lowMemory) {
+							// Only yield the scaled bitmap if the system is not considered to be in low memory mode
+							// On some devices, this bitmap can get up to 50MB due to the large scale factor
```

Comment:
What if we reduce the scale factor in the case of low memory instead of just not yielding anything?

Reply:
We could theoretically do that, but we don't really know how much to scale it down. The scale we use here is the display density divided by the default PDF density of 1/72 inch. We could estimate the bitmap size (which usually is 2 * width * height) and check the free RAM, but `MemoryInfo.availMem` only returns an approximation and part of that value is still occupied memory (see https://developer.android.com/reference/android/app/ActivityManager.MemoryInfo#availMem).
So there's really no 100% way to know if the bitmap would fit in memory or not. Checking for `lowMemory` is a way to prevent OutOfMemoryErrors, even if that means importing the PDF will fail.